### ISSUE TITLE
Setting stack resource limit to infinity via resource.strlimit fails when importing bigframes.pandas on macOS

### ISSUE THREAD

**Issue by sixdimensionalarray** [googleapis/python-bigquery-dataframes#540]
It appears that `resource.strlimit` is unable to set the stack resource limit to `RLIM_INFINITY` in macOS Sonoma 14.1 build 23B74.  

It looks like by running `sudo ulimit -s unlimited` would allow this to be set, but it seems bad to have to escalate to super admin in order to achieve the outcome.  Not sure what the expected outcome is here.  Commenting those lines out lets scripts using bigframes using pandas run, but not sure where the limitation would be hit.

#### Environment details

  - OS type and version: macOS Sonoma version 14.1 build 23B74
  - Shell: zsh
  - Python version: `Python: 3.10.13 (main, Aug 24 2023, 22:36:46) [Clang 14.0.3 (clang-1403.0.22.14.1)]`
  - pip version: `pip 23.2.1 from <path omitted>/.venv/lib/python3.10/site-packages/pip (python 3.10)`
  - `bigframes` version: `bigframes==1.0.0`

```
Python: 3.10.13 (main, Aug 24 2023, 22:36:46) [Clang 14.0.3 (clang-1403.0.22.14.1)]
bigframes==1.0.0
google-cloud-bigquery==3.19.0
ibis==8.0.0
pandas==2.2.1
pyarrow==15.0.1
sqlglot==20.11.0
```

#### Steps to reproduce

  1. pip install bigframes
  2. open a python repl eg. `python3`
  3. import bigframes.pandas as bpd

#### Code example

around L717 of `bigframes/pandas/__init__.py`:

```python
resource.setrlimit(
    resource.RLIMIT_STACK, (resource.RLIM_INFINITY, resource.RLIM_INFINITY)
)
```

```shell
(.venv) $ python3
Python 3.10.13 (main, Aug 24 2023, 22:36:46) [Clang 14.0.3 (clang-1403.0.22.14.1)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import bigframes.pandas as bpd
```

#### Stack trace
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<path-omitted>/.venv/lib/python3.10/site-packages/bigframes/pandas/__init__.py", line 717, in <module>
    resource.setrlimit(
ValueError: not allowed to raise maximum limit
```

**Comment by apb514** [comment]
I’m also experiencing this exact error on MacOS 14.3 and Python 3.9.

**Comment by TrevorBergeron** [comment]
Thank you for reporting this issue. We have merged https://github.com/googleapis/python-bigquery-dataframes/pull/558 to address this bug. This change will be included in the next release of BigFrames - v1.1.0.

**Comment by chelsea-lin** [comment]
Thank you so much for your feedback! I've closed this issue since it's been addressed. If you have any other thoughts or suggestions, please don't hesitate to reopen it.
